lib/parser/prescan.cc

```diff
@@ -116,17 +116,29 @@ void Prescanner::Statement() {
       }
       CHECK(*at_ == '!');
     }
-    EmitChar(tokens, '!');
-    ++at_, ++column_;
-    for (const char *sp{directiveSentinel_}; *sp != '\0';
-         ++sp, ++at_, ++column_) {
-      EmitChar(tokens, *sp);
-    }
-    if (*at_ == ' ') {
-      EmitChar(tokens, ' ');
+    if (directiveSentinel_[0] == '$' && directiveSentinel_[1] == '\0') {
```

Comment:
For a fixed form, column 3 can contain number.

Reply:
Yes.  That works.